### PR TITLE
[BUILD] Check java formatting in Travis CI

### PR DESCRIPTION
**Pull request by m273d15**
* Introduced Formatting stage in travis
* Created script for downloading and executing
  the google java formatter

### CODE REVIEW

**Comment by m273d15** [comment]
See #303 for further information

File: travis/script/format/check_java.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+shopt -s globstar
+
+project_root_dir=$1
+formatter_jar_name=formatter.jar
+
+function download_formatter() {
+  local jar_name=$1
+  local url="https://repo.maven.apache.org/maven2/com/google/googlejavaformat/google-java-format/1.6/google-java-format-1.6-all-deps.jar"
+  echo "Downloading formatter from $url"
+  curl -Lo formatter.jar "$url"
```
Comment (stefaus) [inline travis/script/format/check_java.sh]: I guess you wanted to use jar_name here

File: travis/script/format/check_java.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+function check_formatting() {
+  local jar_name=$1
+
+  echo 'Checking format'
+  java -jar "$jar_name" --dry-run --set-exit-if-changed **/*.java
```
Comment (stefaus) [inline travis/script/format/check_java.sh]: If just check changed files is enough you could maybe use the file list of `git diff master --name-only`.

**Comment by tobous** [comment]
How about also removing the old fotmatting Config files as they are no longer useful and could give users the impression that they are using the correct formatting?

**Comment by m273d15** [comment]
I removed IntelliJ code style settings and the eclipse formatting profile. I did not remove the eclipse clean-up profile, because it configures things like (remove unused imports etc.) and not formatting.

**Comment by fzieris** [comment]
The files in the "patches" folder were always excluded from any auto-format commits. Those are "original" pieces of other projects' code with minor amendments. Auto-formatting these makes it way harder to see the actual "patch".

**Pull request merged**